src/SUFST/src/can_publisher.c

```diff
@@ -51,38 +45,35 @@ UINT queue_send_thread_create(TX_BYTE_POOL* stack_pool_ptr)
                                TX_AUTO_START);
     }
 
-    return ret;
+    return tx_status;
 }
 
 void queue_send_thread_entry(ULONG input)
 {
+    publisher_context_t* publisher_ptr = (publisher_context_t*) input;
+
     // Simulated CAN message
-    rtcan_msg_t queue_data;
-    int ret;
+    rtcan_msg_t* queue_data;
 
     while(1){
     
     // Simulate CAN Message here. TODO: Get input from file.
 
-    queue_data.identifier = CAN_DATABASE_PM100_VOLTAGE_INFO_FRAME_ID;
-    queue_data.length = CAN_DATABASE_PM100_VOLTAGE_INFO_LENGTH;
-
-    for(int i = 0; i<8; i++){
-
-    queue_data.data[i] = i;
+    queue_data->identifier = CAN_DATABASE_PM100_VOLTAGE_INFO_FRAME_ID;
+    queue_data->length = CAN_DATABASE_PM100_VOLTAGE_INFO_LENGTH;
 
+    for(int i = 0; i<8; i++)
+    {
+        queue_data->data[i] = i;
     }
 
     // Send the data to the queue.
-    ret = tx_queue_send(&queue, &queue_data, TX_WAIT_FOREVER);
+    UINT ret = tx_queue_send(&publisher_ptr->tx_queue, &queue_data, TX_WAIT_FOREVER);
     if(ret != TX_SUCCESS){
         return ret;
```

Comment:
This function has `void` return type, so it shouldn't return a value. I'm surprised the compiler doesn't complain about this?

Reply:
@t-bre Just checked and tx_queue_send is not void function. It returns status like other similar functions.

Reply:
![image](https://user-images.githubusercontent.com/47193610/218334599-ec7d1e34-3572-470c-ac22-a787ea6c9f73.png)

Reply:
Yes, I don't mean the ThreadX function I mean your thread function.

Reply:
The line of code with `return ret;`